From: Zebediah Figura <zfigura@codeweavers.com>
Subject: [PATCH] wined3d: Try to load winevulkan.dll instead of vulkan-1.dll.
Message-Id: <20220311215645.783284-1-zfigura@codeweavers.com>
Date: Fri, 11 Mar 2022 15:56:45 -0600

The Khronos Vulkan loader calls into dxgi, which results in infinite recursion.

Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=52640
Signed-off-by: Zebediah Figura <zfigura@codeweavers.com>
---
 dlls/wined3d/adapter_vk.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dlls/wined3d/adapter_vk.c b/dlls/wined3d/adapter_vk.c
index 391225b03a7..44ea5201799 100644
--- a/dlls/wined3d/adapter_vk.c
+++ b/dlls/wined3d/adapter_vk.c
@@ -210,7 +210,8 @@ static BOOL wined3d_load_vulkan(struct wined3d_vk_info *vk_info)
 {
     struct vulkan_ops *vk_ops = &vk_info->vk_ops;
 
-    if (!(vk_info->vulkan_lib = LoadLibraryA("vulkan-1.dll")))
+    if (!(vk_info->vulkan_lib = LoadLibraryA("winevulkan.dll"))
+            && !(vk_info->vulkan_lib = LoadLibraryA("vulkan-1.dll")))
     {
         WARN("Failed to load vulkan-1.dll.\n");
         return FALSE;

-- 
2.35.1

